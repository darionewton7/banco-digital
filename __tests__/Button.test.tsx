import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../src/components/Button';

// Mock onPress function
const mockOnPress = jest.fn();

describe('<CustomButton />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<CustomButton title="Test Button" onPress={mockOnPress} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const tree = renderer.create(<CustomButton title="Disabled Button" onPress={mockOnPress} disabled />).toJSON();
    expect(tree).toMatchSnapshot();
    // Check if opacity style is applied for disabled state
    // Note: Specific style checks might be brittle, snapshot is often preferred
  });

  it('renders correctly when loading', () => {
    const tree = renderer.create(<CustomButton title="Loading Button" onPress={mockOnPress} loading />).toJSON();
    expect(tree).toMatchSnapshot();
    // Check if ActivityIndicator is rendered
  });

  it('renders correctly with secondary variant', () => {
    const tree = renderer.create(<CustomButton title="Secondary Button" onPress={mockOnPress} variant="secondary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with danger variant', () => {
    const tree = renderer.create(<CustomButton title="Danger Button" onPress={mockOnPress} variant="danger" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // We can't easily test onPress directly with react-test-renderer
  // For interaction tests, consider using @testing-library/react-native
});

