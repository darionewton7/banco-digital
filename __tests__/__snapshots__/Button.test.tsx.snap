// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> renders correctly when disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6c757d",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 16,
      "minHeight": 48,
      "opacity": 0.6,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Disabled Button
  </Text>
</View>
`;

exports[`<CustomButton /> renders correctly when loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0056b3",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 16,
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <ActivityIndicator
    color="#ffffff"
  />
</View>
`;

exports[`<CustomButton /> renders correctly with danger variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#dc3545",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 16,
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Danger Button
  </Text>
</View>
`;

exports[`<CustomButton /> renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0056b3",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 16,
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Test Button
  </Text>
</View>
`;

exports[`<CustomButton /> renders correctly with secondary variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6c757d",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 16,
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Secondary Button
  </Text>
</View>
`;
